src/main/scala/analysis/UtilMethods.scala

```diff
@@ -0,0 +1,52 @@
+package analysis.eval
+import ir.*
+import analysis.solvers.*
+import analysis.Lattice
+import analysis.*
+import util.Logger
+
+/** Evaluate an expression in a hope of finding a global variable.
+  *
+  * @param exp
+  *   : The expression to evaluate (e.g. R1 + 0x1234)
+  * @param n
+  *   : The node where the expression is evaluated (e.g. mem[R1 + 0x1234] <- ...)
+  * @return:
+  *   The evaluated expression (e.g. 0x69632)
+  */
+def evaluateExpression(exp: Expr, n: CfgNode, constantProp: Map[CfgNode, Map[Variable, Any]]): Expr = {
```

Comment:
Why not pass in the constant propagation results just for the node we're evaluating? Is there a reason this method only supports such a limited amount of operators? Is it not possible to build a more type-robust evaluator that will only deal with the operators that have bitvector inputs & outputs so there aren't casts everywhere surrounding its usage?